api/chat.go

```diff
@@ -260,15 +267,26 @@ func getMessages(c *gin.Context) {
 	c.JSON(http.StatusOK, chats)
 }
 
+func getUsers(c *gin.Context) {
+	_, exists := c.Get("TUMLiveContext")
+	if !exists {
+		c.AbortWithStatus(http.StatusNotFound)
+		return
+	}
+	users := userStorage.GetAll()
```

Comment:
Just a thought without understanding your code in detail yet:

Why not just allow @-ing users who already sent a message to the chatroom, this way we can use the chats table to get them (`select u.* from chats join users u on chats.user_id = u.id where stream_id = ...`) and don't need the `userStorage`, right? (Feel free to make the `UserID` in the Chat table a proper foreign key if that would make things easier)

Reply:
That is a smart idea. I guess I sometimes like to make things more complicated than they need to be 😆

Reply:
Me too 😀 a second pair of eyes always helps when I start over engineering